### ISSUE TITLE
Implement pipeline/03_embed_corpus.py

### ISSUE THREAD

**Issue by atefft** [atefft/movie-semantic-search#7]
## What
Create `pipeline/03_embed_corpus.py`. The script must expose the following module-level functions (required by the test suite in `pipeline/tests/test_03_embed_corpus.py`) and a `main()` entry point that drives the full embedding pipeline.

Functions to implement:
- `parse_release_year(raw: str) -> int | None` — accepts a YYYY-MM-DD date string; returns `None` for empty input or any value that does not match that format
- `extract_genres(genres_json: str) -> list[str]` — parse the JSON map `{"/m/xxx": "Drama", ...}` and return the values as a list
- `load_metadata(path) -> dict[str, dict]` — read `movie.metadata.tsv` (tab-separated, no header, 9 columns), return dict keyed by `wikipedia_movie_id` (col 0); each value has keys `title` (col 2), `movie_release_date` (col 3), `movie_genres` (col 8)
- `load_summaries(path) -> dict[str, str]` — read `plot_summaries.txt` (tab-separated, 2 cols), return dict keyed by `wikipedia_movie_id`
- `join_data(metadata_map, summaries_map) -> list[dict]` — inner join on key; each record: `movie_id`, `title`, `release_year` (from `parse_release_year`), `genres` (from `extract_genres`), `summary_snippet` (first 300 chars of summary)

`main()` logic:
1. Parse CLI args: `--batch-size` (default 64), `--triton-host` (default `localhost`), `--triton-port` (default `8001`)
2. Load and join data from `data/raw/movie.metadata.tsv` and `data/raw/plot_summaries.txt`
3. Send one warmup request to Triton gRPC before the first batch
4. Send records in batches of `--batch-size` to Triton gRPC at `{triton_host}:{triton_port}`, collecting float32 embeddings
5. Write `data/embeddings/embeddings.npy` (float32 array, shape `[N, 384]`) and `data/embeddings/metadata.json` (JSON array in same order as embeddings)
6. Idempotent: if both output files already exist, print a message and exit

Also add `tritonclient[grpc]` and `tqdm` to `pipeline/requirements.txt`.

## Why
Satisfies acceptance criteria: "`pipeline/03_embed_corpus.py` exists and reads `data/raw/movie.metadata.tsv` and `data/raw/plot_summaries.txt`, joins on `wikipedia_movie_id`, sends batch embeddings via Triton gRPC, and writes `data/embeddings/embeddings.npy` and `data/embeddings/metadata.json`" — from feature #2.

## Files to change
- `pipeline/03_embed_corpus.py` — create with all functions and `main()` described above
- `pipeline/requirements.txt` — add `tritonclient[grpc]` and `tqdm` lines

## Implementation notes
Follow the pattern from `pipeline/02_export_model.py`: shebang `#!/usr/bin/env python3`, module-level constants, a `main()` function, `if __name__ == "__main__": main()`.

Triton gRPC call pattern (using `tritonclient.grpc`):
```python
import tritonclient.grpc as grpcclient
import numpy as np

client = grpcclient.InferenceServerClient(url=f"{host}:{port}")

# Tokenize text into input_ids, attention_mask, token_type_ids (int32, shape [B, 128])
inputs = [
    grpcclient.InferInput("input_ids", [B, 128], "INT32"),
    grpcclient.InferInput("attention_mask", [B, 128], "INT32"),
    grpcclient.InferInput("token_type_ids", [B, 128], "INT32"),
]
inputs[0].set_data_from_numpy(input_ids)
inputs[1].set_data_from_numpy(attention_mask)
inputs[2].set_data_from_numpy(token_type_ids)

outputs = [grpcclient.InferRequestedOutput("token_embeddings")]
result = client.infer(model_name="all-minilm-l6-v2", inputs=inputs, outputs=outputs)
token_embeddings = result.as_numpy("token_embeddings")  # [B, seq, 384]
# Mean-pool over sequence dimension using attention_mask
mask = attention_mask[:, :, None].astype(float)
pooled = (token_embeddings * mask).sum(axis=1) / mask.sum(axis=1)  # [B, 384]
```

Tokenizer: load `sentence-transformers/all-MiniLM-L6-v2` with `AutoTokenizer.from_pretrained`, padding to max_length=128.

TSV column indices for `movie.metadata.tsv`: 0=wikipedia_movie_id, 2=movie_name, 3=movie_release_date, 8=movie_genres.

The conftest stubs `tritonclient`, `tritonclient.grpc`, `tqdm`, etc. so all unit tests run without those packages installed.

## Unit tests
Tests already exist in `pipeline/tests/test_03_embed_corpus.py`. They test:
- `parse_release_year("2000-11-22")` → `2000`
- `parse_release_year("1979")` → `None` (bare year is not YYYY-MM-DD)
- `parse_release_year("")` → `None`
- `parse_release_year("unknown")` → `None`
- `parse_release_year("2005-03")` → `None` (YYYY-MM is not YYYY-MM-DD)
- `extract_genres('{"k": "Drama", "k2": "Adventure"}')` → `["Adventure", "Drama"]`
- `extract_genres("{}")` → `[]`
- `load_metadata(path)` keyed by movie_id, `result["111"]["title"] == "Cast Away"`, 4 rows total
- `load_summaries(path)` keyed by movie_id, `"FedEx" in result["111"]`
- `join_data(...)` excludes movie_id "333" (no summary), excludes "555" (no metadata), includes "111" and "222"
- Each record has fields: `movie_id`, `title`, `release_year`, `genres` (list), `summary_snippet`
- `release_year` for "111" → `2000`; for empty date → `None`
- `summary_snippet` capped at 300 chars; short summary kept as-is

## Acceptance criteria
- [ ] `python3 -m pytest pipeline/tests/test_03_embed_corpus.py -v` passes with 0 failures (no skips)
- [ ] `pipeline/03_embed_corpus.py --help` shows `--batch-size`, `--triton-host`, `--triton-port` arguments
- [ ] `pipeline/requirements.txt` contains `tritonclient[grpc]` and `tqdm`
- [ ] All unit tests pass

## Test command
```bash
python3 -m pytest pipeline/tests/ -v
```

## Blocked by
none

## Parent feature
Part of #2

**Comment by atefft** [comment]
PR opened: https://github.com/atefft/movie-semantic-search/pull/29
